### PR TITLE
Add isAtLeast(OS) method

### LINKED ISSUE #3: Is (version) or greater helper methods
Feature request:  a common version use case is determining a minimum and/or maximum range of versions.  It would be useful for you to add a few helper methods that do not require an explicit list.

For example, you currently have `public boolean is(OS... operatingSystems)`.   I'd suggest adding `public boolean isAtLeast(OS operatingSystem)`.

### PR DESCRIPTION
**Pull request by dbwiddis**
Fixes #3 

Makes the assumption the "Unknown" versions are newer, which might not be true but is more likely to be true than not.

Used your same formatting with tabs, which are evil. ;)

### CODE REVIEW

File: src/main/java/io/github/cegredev/josi/OS.java [hunk 1]
```diff
@@ -582,6 +582,22 @@ public boolean is(OS... operatingSystems) {
...
 
+	/**
+	 * Checks if this OS is at the same or later version of the given OS. Only relevant for Windows and Mac.
+	 *
+	 * @param operatingSystem An operating system with version.
+	 * @return If this OS is the same version as the given OS or a later one of the same family.
+	 */
+	public boolean isAtLeast(OS operatingSystem) {
+		// Restrict to families with versions
+		enforceFamily(Family.MAC, Family.WINDOWS);
```
Comment (dbwiddis) [inline src/main/java/io/github/cegredev/josi/OS.java:593]: I realized later perhaps we can allow other OS's to equal each other if the family matches so this should probably be changed to `enforceNotFamily(Family.UNKNOWN)`

File: src/test/java/io/github/cegredev/josi/OSCheckTests.java [hunk 2]
```diff
@@ -39,6 +39,25 @@ public void testIsOS() {
...
+		String message = "OS was not recognized!";
+		assertTrue(OS.WIN_10.isAtLeast(OS.WIN_95), message);
+		assertTrue(OS.WIN_7.isAtLeast(OS.WIN_7), message);
+		assertFalse(OS.WIN_8.isAtLeast(OS.WIN_10), message);
+		assertTrue(OS.MAC_OS_BIG_SUR.isAtLeast(OS.MAC_OS_SIERRA), message);
+		assertTrue(OS.MAC_OS_CATALINA.isAtLeast(OS.MAC_OS_CATALINA), message);
+		assertFalse(OS.MAC_OSX_LION.isAtLeast(OS.MAC_OSX_MOUNTAIN_LION), message);
+
+		String notThrowMessage = "Did not throw even though OS was not allowed!";
+		assertThrows(UnsupportedOSException.class, () -> OS.UBUNTU.isAtLeast(OS.UBUNTU),
```
Comment (dbwiddis) [inline src/test/java/io/github/cegredev/josi/OSCheckTests.java:53]: This should maybe be changed to be "true".

**Review by cegredev — CHANGES_REQUESTED** [review]
This is a great idea, thanks a lot!

One thing I'd like to change is the fact that you have those `enforce` calls inside `isAtLeast`. I can understand why one would want to throw an exception in those cases but in my opinion that is overkill in this scenario and just returning `false` would be enough. I can see this method being used instead of `OS.isFamily` and we wouldn't want that throwing an exception if the result was false.

Also, based on the other helper methods the user doesn't expect one starting with "is" to throw an exception. That's what "enforce" is for. For this reason I'd actually like to keep the code you wrote, but move it to a separate function `public void enforceAtLeast(OS operatingSystem)` and have it throw an exception if the result of `compareTo` is smaller than 0.

**Comment by cegredev** [comment]
> So two methods; the `isAtLeast()` just returning false on family mismatch (or unknown family) or incorrect version alignment, and `enforceAtLeast()` doing similar but with exceptions (and probably just calling `isAtLeast()` internally)?

Yes, exactly! :)

**Comment by cegredev** [comment]
About this: I'm not sure if you missed my last comment...

> Yes, but that is only the case if you just consider the family of the operating system. Take Solaris for example. As of right now, it is the only member of the OTHER family (besides UNKNOWN of course). If I were to say OS.SOLARIS.isAtLeast(OS.SOLARIS) that would return false, even though that's clearly wrong.
> 
> I agree that OS.UNKNOWN and OS.UNKNOWN should return false, but not OS.Family.OTHER and OS.Family.OTHER.

...but that's the reason I haven't merged this yet. I'd have no problem merging and then implementing this little fix myself, but before I do that I wanted to make sure I don't go over your head, in case you do actually want to do this yourself or disagree.

**Comment by dbwiddis** [comment]
I think I got confused by UNKNOWN (OS) vs. OTHER (Family).   Really the correct solution is to put SOLARIS in a UNIX family along with AIX (which currently matches into LINUX).  Although you can change that conditional to just an UNKNOWN OS.  I'm fine if you want to tweak the logic after you merge (or even push to the PR fork before you merge... if you haven't learned how to do that as a maintainer, you should! )

**Comment by cegredev** [comment]
> put SOLARIS in a UNIX family along with AIX

Yep, I'll implement that soon.

> push to the PR fork before you merge... if you haven't learned how to do that as a maintainer, you should!

Just did that. I really am trying my best to do a good job of this, even though I'm new to all of it, believe me. I look forward to learning a lot more as this project progresses, haha.

**Comment by dbwiddis** [comment]
> I look forward to learning a lot more as this project progresses, haha.

It's really one of the best places out there to self-learn software engineering!  Try getting hired as a senior software engineer based primarily on 7 years of maintaining an open source project. ;)

**Comment by cegredev** [comment]
Haha, sounds like a good plan, see you in 7 years then ;)

**Pull request merged**